fastlane/spec/actions_specs/spm_spec.rb

```diff
@@ -339,6 +339,43 @@
           expect(result).to eq("set -o pipefail && swift package --verbose generate-xcodeproj --xcconfig-overrides Package.xcconfig 2>&1 | xcpretty --simple")
         end
       end
+
+      context "when simulator is specified" do
+        it "adds simulator flags to the build command" do
+          result = Fastlane::FastFile.new.parse("lane :test do
+              spm(
+                command: 'build',
+                simulator: 'iphonesimulator'
+              )
+            end").runner.execute(:test)
+
+          expect(result).to eq("swift build -Xswiftc -sdk -Xswiftc $(xcrun --sdk iphonesimulator --show-sdk-path) -Xswiftc -target -Xswiftc x86_64-apple-ios$(xcrun --sdk iphonesimulator --show-sdk-version | cut -d '.' -f 1)-simulator")
+        end
+
+        it "raises an error if simulator syntax is invalid" do
+          expect do
+            Fastlane::FastFile.new.parse("lane :test do
+              spm(
+                command: 'build',
+                simulator: 'invalid_simulator'
+              )
+            end").runner.execute(:test)
+          end.to raise_error("Invalid simulator syntax. Please use 'iphonesimulator', 'ipadsimulator', or 'macossimulator'.")
```

Comment:
`ipadsimulator` is not part of `self.valid_simulators` 👀  should it be?

Reply:
Good catch. Done.